### PR TITLE
Migrate Wire to use composite builds

### PR DESCRIPTION
**Pull request by oldergod**
You can read about composite builds [here](https://docs.gradle.org/current/userguide/composite_builds.html)

The project would now be split in two
1. The root (still called wire) which contains what we don't publish, usually all our tests.
2. The `wire-library` which is all artifacts we publish (except for test-utils)

`wire-library` is self-contained, and thanks to composite builds, the `root` project can depend on `wire-library` as if it was externally published. This means we can use the Wire Gradle Plugin® *built from the source*. :applause:

There should be *no* behavior change with this PR.

Left tasks:
- [ ] make sure the publish script isn't broken.
- [ ] make sure the mkdoc thing isn't broken.

### CODE REVIEW

File: local.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
```
Comment (Egorand) [inline local.properties]: Add to gitignore?

File: wire-tests/build.gradle [hunk 2]
```diff
@@ -8,7 +8,7 @@ kotlin {
         kotlin.srcDirs += 'src/jvmKotlinInteropTest/proto-kotlin'
         dependencies {
           implementation(compilations.main.compileDependencyFiles)
-          implementation project(':wire-grpc-client')
+          implementation "com.squareup.wire:wire-grpc-client:3.2.0-SNAPSHOT"
```
Comment (Egorand) [inline wire-tests/build.gradle]: Should this reference coordinates inside `deps`?

Comment (Egorand) [inline wire-tests/build.gradle]: Should this reference coordinates inside `deps`?

Comment (oldergod) [reply wire-tests/build.gradle]: yes!

Comment (oldergod) [reply wire-tests/build.gradle]: yes!

File: local.properties (hunk 1 above)
Comment (oldergod) [reply local.properties]: fixed

**Comment by oldergod** [comment]
Actually failing because of https://youtrack.jetbrains.com/issue/KT-36784

**Pull request merged**